x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/persistence/JobResultsProvider.java

```diff
@@ -525,7 +525,8 @@ private SearchRequestBuilder createLatestDatafeedTimingStatsSearch(String indexN
             .setSize(1)
             .setIndicesOptions(IndicesOptions.lenientExpandOpen())
             .setQuery(QueryBuilders.idsQuery().addIds(DatafeedTimingStats.documentId(jobId)))
-            .addSort(SortBuilders.fieldSort(DatafeedTimingStats.TOTAL_SEARCH_TIME_MS.getPreferredName()).order(SortOrder.DESC));
+            .addSort(SortBuilders.fieldSort(DatafeedTimingStats.TOTAL_SEARCH_TIME_MS.getPreferredName())
+                .unmappedType("double").order(SortOrder.DESC));
```

Comment:
@droberts195 there is also a sort on `bucket_count` in this file. Is that one going to cause problems as well? I am currently trying to find it in the mappings.

Reply:
Ah, it looks like the field is there from `DataCounts` previously.

Reply:
Good thought.  I just double checked and it doesn't, because `bucket_count` is also a field in `data_counts` documents that have existed since 5.4.  So it would have the correct mappings in any results index that has been created by any released version of ML.  Of course, it could break the job stats endpoint in the case of a results index created manually with incomplete mappings, but we have many sorts where that could happen.  Since this PR is green I'd rather merge as-is to fix the critical problem.  If we decide we want to make _all_ our sorts more robust to incomplete mappings we can do that in a followup.